python/pyspark/sql/types.py

```diff
@@ -1681,38 +1681,53 @@ def from_arrow_schema(arrow_schema):
          for field in arrow_schema])
 
 
-def _check_series_convert_date(series, data_type):
-    """
-    Cast the series to datetime.date if it's a date type, otherwise returns the original series.
+def _arrow_column_to_pandas(column, data_type):
+    """ Convert Arrow Column to pandas Series.
+
+    If the given column is a date type column, creates a series of datetime.date directly instead
+    of creating datetime64[ns] as intermediate data.
```

Comment:
minor: I think these details belong as a comment internally rather than in the doc string.

Reply:
It would be nice to say that for dates this will return `datetime.date`, but yeah maybe move the part about datetime[64] as intermediate to an internal comment.  `_arrow_table_to_pandas` has a comment that the reason for this is to match pyspark w/o arrow, but maybe it would be good to add here as well.